### PR TITLE
Make `yxi` a standalone tool

### PR DESCRIPTION
**Pull request by rachitnigam**
See it in action:
```
cargo run --manifest-path tools/yxi/Cargo.toml tests/correctness/seq.futil
```

### CODE REVIEW

**Comment by rachitnigam** [comment]
@nathanielnrn does this look reasonable? If so, do you want to take over and move around the scripts to use the right tool binary (`yxi` instead of `calyx -b ...`).

**Comment by nathanielnrn** [comment]
Sure, will do.

**Comment by ayakayorihiro** [comment]
This is great, thanks! I confirmed that this YXI tool works for both `ref` and `@external` memories, and we won't need to require users to compile with `--features yxi` :) When I get a chance I'll make changes to fud2!!

File: fud2/src/lib.rs [hunk 1]
```diff
@@ -282,8 +282,7 @@ pub fn build_driver(bld: &mut DriverBuilder) {
         e.build_cmd(&[only_refs_calyx], "external-to-ref", &[input], &[])?;
 
         // Get YXI to generate JSON for testbench generation
-        e.build_cmd(&[memories_json], "calyx", &[only_externals_calyx], &[])?;
-        e.arg("backend", "yxi")?;
+        e.build_cmd(&[memories_json], "yxi", &[only_externals_calyx], &[])?;
```
Comment (nathanielnrn) [inline fud2/src/lib.rs:285]: Tried to address the fud2 changes for firrtl as well @ayakayorihiro. Lmk if this looks ok?

Comment (ayakayorihiro) [reply fud2/src/lib.rs:285]: This looks great, thanks!!

**Comment by nathanielnrn** [comment]
Merging! This should close #1932

**Comment by ethanuppal** [comment]
Can we remove `struct YxiBackend` since it's no longer referenced anywhere? Sorry for commenting on a closed PR but I don't know where else to put this.

image

**Comment by nathanielnrn** [comment]
Yeah it'd be great if you could do that, apologies for missing this

**Comment by ethanuppal** [comment]
I fixed it on https://github.com/calyxir/calyx/tree/ethan-yxi, where I'm also rewriting what you did so I can figure out how it works (not to replace it) (and also to use `@fast`)

**Pull request merged**